lib/trip.rb

```diff
@@ -8,14 +8,68 @@ def initialize(input)
       @id = input[:id]
       @driver = input[:driver]
       @passenger = input[:passenger]
-      @start_time = input[:start_time]
-      @end_time = input[:end_time]
-      @cost = input[:cost]
-      @rating = input[:rating]
+      @start_time = check_start_time(input[:start_time])
+      @end_time = check_end_time(input[:end_time])
+      @cost = check_cost(input[:cost])
+      @rating = check_rating(input[:rating])
 
-      if @rating > 5 || @rating < 1
-        raise ArgumentError.new("Invalid rating #{@rating}")
+      check_time
+
+    end
+
+    def duration
+      @end_time - @start_time
+    end
+
+    def inspect
+      "#<#{self.class.name}:0x#{self.object_id.to_s(16)}>"
+    end
+
+    private
+
+    def check_cost(cost_input)
+      if cost_input == nil
+        return nil
+      else
+        return cost_input.to_f
+      end
+    end
+
+    def check_rating(rating_input)
```

Comment:
To avoid nested `if` statements, let me introduce the concept of a precondition. Often we have preconditions for a section of code to run. In this case, if `rating_input` is `nil`, we really can't proceed with anything else.

We can instead do what is called a short-circuit return, where we `return` prematurely in the method.

Here's what it would look like:

```ruby
    def check_rating(rating_input)
      return nil if rating_input.nil?

      if rating_input > 5 || rating_input < 1
        raise ArgumentError.new("Invalid rating #{@rating}")
      else
        @rating = rating_input
      end
    end
```

Reply:
Another benefit: you can see all the preconditions at the top, so it's a nice organizational tool as well.